### PR TITLE
Lingo conditional

### PR DESCRIPTION
**Pull request by Coolrunner86**
Added basis of the preprocessor into Lingo parser. Conditional grammar compilation.

### CODE REVIEW

**Comment by alstrup** [comment]
You have to add documentation to doc/lingo.markdown

**Comment by Coolrunner86** [comment]
> You have to add documentation to doc/lingo.markdown

Done

File: doc/lingo.markdown [hunk 1]
```diff
@@ -151,6 +151,31 @@ The grammar
...
+	prod = 'a'* { A() };
+	#elif CONDITION_2
+	prod = 'b'* { B() };
+	#else
+	prod = 'c'* { C() };
+	#endif
+
+Nested conditions are also allowed.
+Definitions can be stated through `definitions` parameter for pegcompiler or by using `compilePegGrammar3` function.
+
```
Comment (alstrup) [inline doc/lingo.markdown]: It is not clear whether you can do

prod = something
#ifdef CONDITION
    | more
#else
#endif

Or is it only for complete productions?

Comment (Coolrunner86) [reply doc/lingo.markdown]: It's only for complete production. But if there is a need in such functionality, it can be implemented.

Comment (alstrup) [reply doc/lingo.markdown]: Then document that limitation, and then OK to merge.

Comment (Coolrunner86) [reply doc/lingo.markdown]: In fact, this is rather middle-processor than preprocessor. We could implement stand-alone version of it and use it for other file types.

**Comment by alstrup** [comment]
Yes, that was my original thought, and that seems to be much simpler.

**Pull request closed without merging**